### DISCUSSION TITLE
Swup 4.3.4 + Gia Plugin 1.0.1

### DISCUSSION THREAD

**Discussion by cubachatou** [swup/swup#784] [Help] [answered]
Hello there!

Upgrading my old version of swup.js by following the instruction on the website I encountered the following problem:

`Uncaught (in promise) TypeError: this.swup.on is not a function`

Accordingly to the new documentation `swup.on('contentReplaced'); swup.on('willReplaceContent'); swup.off('contentReplaced'); swup.off('willReplaceContent')` were changed but the seems like swup-gia-plugin use the old ones, here's the part of index.js of GiaPlugin:

__createClass(GiaPlugin, [{
		key: 'mount',
		value: function mount() {
			if (this.options.firstLoad) {
				this.mountComponents();
			}
			this.swup.on('contentReplaced', this.mountComponents);
			this.swup.on('willReplaceContent', this.unloadComponents);
		}
	}, {
		key: 'unmount',
		value: function unmount() {
			this.unloadComponents();
			this.swup.off('contentReplaced', this.mountComponents);
			this.swup.off('willReplaceContent', this.unloadComponents);
		}
	}]);_

And here's my init code that was working fine before I upgraded the swup:

_const swup = new Swup({
	linkSelector: 'a[href^="' + window.location.origin + '"]:not([data-no-swup]), a[href^="/"]:not([data-no-swup]), a[href^="./"]:not([data-no-swup]), a[href^="#"]:not([data-no-swup])',
	plugins: [
		new SwupGiaPlugin({
			animationScope: 'containers',
			components: {...},
			log: false,
		}),
		new SwupBodyClassPlugin(),
	],
});_

Any solutions?

**Comment by daun** [comment]
Hi 🧃 It looks like you're using the new swup version but including an old version of gia-plugin. If you update gia-plugin to the latest version 2.0.1 as well, it should work as expected.

**Reply by cubachatou** [reply]
Thanks for the answer!
Where can I find the latest version? NPM shows 1.0.1 is the latest version.

**Reply by daun** [reply, marked as answer]
[npm is showing 2.0.1 as the latest version for me](https://www.npmjs.com/package/@swup/gia-plugin?activeTab=versions). Does it show something different to you?

**Reply by cubachatou** [reply]
My bad! Just my inattention.

**Reply by daun** [reply]
No worries!
